quinn-proto/src/bytes_source.rs

```diff
@@ -0,0 +1,231 @@
+//! Defines types which can be lazily converted into `Bytes` chunks
+
+use bytes::Bytes;
+
+/// A source of one or more buffers which can be converted into `Bytes` buffers on demand
+///
+/// The purpose of this data type is to defer conversion as long as possible,
+/// so that no heap allocation is required in case no data is writable.
+pub trait BytesSource {
+    // Returns the total amount of bytes to write
+    fn len(&self) -> usize;
```

Comment:
Is this method used anywhere outside of tests? If not, can we get rid of it?

Reply:
I guess it can be removed. It kind of comes for free and it seems like for anything bytes-like a length makes sense. But if there are no consumers it might indeed to better to leave it out.